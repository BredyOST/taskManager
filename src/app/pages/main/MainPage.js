import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { HomeCoverDiv, HomeWrapDiv } from './styled/mainStyled.ts';
import AddTask from '../../../features/addTask/addTask.tsx';
import TaskLists from '../../../features/taskLists/taskLists.tsx';
import SwitchTasksCategory from '../../../features/switchTasksCategory/switchTasksCategory.tsx';
import React from 'react';
const MainPage = () => {
    const [openList, setOpenList] = React.useState(true);
    const showListsTasks = React.useCallback(() => {
        setOpenList((prev) => !prev);
    }, [openList]);
    return (_jsx(HomeWrapDiv, { children: _jsxs(HomeCoverDiv, { children: [_jsx(AddTask, { openList: openList, showListsTasks: showListsTasks }), _jsx(TaskLists, { openList: openList }), _jsx(SwitchTasksCategory, {})] }) }));
};
export default MainPage;
