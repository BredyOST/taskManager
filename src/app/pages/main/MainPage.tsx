import { HomeCoverDiv, HomeWrapDiv } from './styled/mainStyled.ts';
import AddTask from '../../../features/addTask/addTask.tsx';
import TaskLists from '../../../features/taskLists/taskLists.tsx';
import SwitchTasksCategory from '../../../features/switchTasksCategory/switchTasksCategory.tsx';
import React from 'react';
import { ForFunc } from '../../../shared/types/types.ts';

const MainPage = () => {
    const [openList, setOpenList] = React.useState<boolean>(true);

    const showListsTasks: ForFunc<void, void> = React.useCallback(() => {
        setOpenList((prev) => !prev);
    },[openList]);

    return (
        <HomeWrapDiv>
            <HomeCoverDiv>
                <AddTask openList={openList} showListsTasks={showListsTasks} />
                <TaskLists openList={openList} />
                <SwitchTasksCategory />
            </HomeCoverDiv>
        </HomeWrapDiv>
    );
};

export default MainPage;
