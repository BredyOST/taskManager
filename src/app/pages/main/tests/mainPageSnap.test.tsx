import {render} from "@testing-library/react";
import MainPage from "../MainPage.tsx";

jest.mock('../../../../features/taskLists/taskLists.tsx', () => () => <div>TaskLists</div>)
jest.mock('../../../../features/addTask/addTask.tsx', () => () => <div>AddTask</div>)
jest.mock('../../../../features/switchTasksCategory/switchTasksCategory.tsx', () => () => <div>SwitchTasksCategory</div>)

describe('Mainpage', () => {
    test('mathes snapshot', () => {
        const {asFragment} = render(<MainPage/>)
        expect(asFragment()).toMatchSnapshot()
    })
})