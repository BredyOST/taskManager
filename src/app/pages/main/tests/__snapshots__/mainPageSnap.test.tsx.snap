// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainpage mathes snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb cuPcaF"
  >
    <div
      class="sc-gtLWhw eWNOFE"
    >
      <div>
        AddTask
      </div>
      <div>
        TaskLists
      </div>
      <div>
        SwitchTasksCategory
      </div>
    </div>
  </div>
</DocumentFragment>
`;
